src/OrchardCore.Modules/OrchardCore.Admin/AdminActionConstraint.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using Microsoft.AspNetCore.Mvc.ActionConstraints;
+
+namespace OrchardCore.Admin
+{
+    public class AdminActionConstraint : IActionConstraint
+    {
+        public int Order => 0;
+
+        public bool Accept(ActionConstraintContext context)
+        {
+            if (context.RouteContext.HttpContext.Request.Path.Value.StartsWith("/OrchardCore", StringComparison.OrdinalIgnoreCase))
```

Comment:
I would actually test that it doesn't start with the expected admin prefix from the options

Reply:
yeah that makes sense, and can work with `.StartsWithSegments` then. done